corehq/apps/ivr/migrations/0002_call_app_id.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.22 on 2019-10-28 20:52
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+from corehq.apps.ivr.models import Call
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.util.django_migrations import skip_on_fresh_install
+
+
+app_id_by_form_unique_id = {}
+
+
+def _update_model(model, domain):
+    if model.form_unique_id is None:
+        return None
+
+    if model.form_unique_id not in app_id_by_form_unique_id:
+        apps = get_apps_in_domain(domain)
+        for app in apps:
+            for module in app.modules:
+                for form in module.get_forms():
+                    app_id_by_form_unique_id[form.unique_id] = app.get_id
+
+    model.app_id = app_id_by_form_unique_id.get(model.form_unique_id, None)
+    if model.app_id:
+        model.save()
+
+
+@skip_on_fresh_install
+def _populate_app_id(apps, schema_editor):
+    for call in Call.objects.filter(form_unique_id__isnull=False).all():
```

Comment:
I believe the [idiomatic](https://docs.djangoproject.com/en/2.2/ref/migration-operations/#django.db.migrations.operations.RunPython) way to access Django models in a migration is
```python
Call = apps.get_model('ivr', 'Call')
```
That's so the migration code is still valid even if the model changes in the future or is deleted.

Reply:
Thanks, I didn't know that. 5dd2681db42

Reply:
Moved this change over to https://github.com/dimagi/commcare-hq/pull/25757/, so now it's 2ace5a83f6cf1259c1d11c793d6c2a2ccf9f847f